packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -482,6 +482,11 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
         enableDrag: enableDrag,
       ),
     );
+    if (isScrollControlled)
```

Comment:
Instead of removing the padding and then adding it back, could line#474 above be 
```dart
removeTop: !isScrollControlled
```

Reply:
That would be an option although it does have a different effect.
With the version in this PR the padding is handled by the bottomsheet route.
If we would change it to remove top false for isScrollControlled than every user of the bottomsheet needs to account for it themselves (but only for the scroll controlled ones)

Reply:
Ah I see, thank you for explaining!